x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateFormatTestScript.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
+
+import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
+
+import org.elasticsearch.core.PathUtils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class DateFormatTestScript {
+
+    private static class TestRecord {
+        private final LocalDateTime localDateTime;
+        private final String formatString;
+
+        TestRecord(LocalDateTime localDateTime, String formatString) {
+            this.localDateTime = localDateTime;
+            this.formatString = formatString;
+        }
+    }
+
+    private static final List<String> PATTERNS = new ArrayList<>(DateFormatter.FORMATTER_MAP.keySet());
+    private static final List<TestRecord> TESTRECORDS = new ArrayList<>();
+
+    private static final Random RANDOM = new Random();
```

Comment:
Please, use `org.elasticsearch.common.Randomness.get()` here

Reply:
I tried to use the method org. elasticsearch. common.Randomness. get(), but it caused an error,
so I used a solution similar to the one in ToCharTestScript.java